File: Language Exercises/C_C--master/stack.cpp
#include <iostream>
int arr[0];
class StackX
{
	int maxSize;
	
	int top;

public:

	StackX(int s)
	{
		maxSize = s;
		top = -1;
		arr[maxSize];
	}


	void push(int j)
	{
		arr[++top] = j;
	}

	int pop()
	{
		return arr[top--];
	}

	int peek()
	{
		return arr[top];
	}

	bool isEmpty()
	{
		return (top==-1);
	}

	bool isFull()
	{
		return (top == (maxSize-1));
	}

	void display()
	{
		
	}
};

int main()
{
	StackX stacker(6);
	stacker.push(10);
	stacker.push(20);
	stacker.push(30);
	stacker.push(40);
	stacker.push(50);
	stacker.push(60);
	

	for(int i = 0;i<6;i++)
		cout<<arr[i]<<" ";
		//double value = stacker.pop();
		
	
	cout<<endl;
	stacker.pop();
	cout<<stacker.peek();

	cin.get();
	return 0;

}

File: language-exercises/java-practice/stack.java
public class stack
{
	int maxSize;
	long[] stackArray;
	int top;

	public stack(int s)
	{
		maxSize = s;
		stackArray = new long[maxSize];
		top = -1;
	}

	public void push(long j)
	{
		stackArray[++top] = j;
	}

	public long pop()
	{
		return stackArray[top--]; 
	}

	public long peek()
	{
		return stackArray[top];
	}

	public boolean isEmpty()
	{
		return (top == -1);
	}

	public boolean isFull()
	{
		return (top == maxSize-1);
	}

	public static void main(String args[])
	{
		stack stacks = new stack(10);
		stacks.push(10);
		stacks.push(20);
		stacks.push(30);
		stacks.push(40);

		while(!stacks.isEmpty())
		{
			long value = stacks.pop();
			System.out.println(value);
			System.out.println(" ");
		}

		System.out.println(" ");

	}
			
}